pkg/controller/.import-restrictions

```diff
@@ -253,7 +253,7 @@
         "k8s.io/kubernetes/pkg/capabilities",
         "k8s.io/kubernetes/pkg/master/ports",
         "k8s.io/kubernetes/pkg/scheduler/api",
-        "k8s.io/kubernetes/pkg/scheduler/util",
+        "k8s.io/kubernetes/pkg/scheduler/internal",
```

Comment:
This breaks a logical dependency contract.  The controller and scheduler are distinct.  Having the controller depend on an internal package in the scheduler means the two can never be separated.  This is a bad outcome.  We should not make this change.

Reply:
> This breaks a logical dependency contract. The controller and scheduler are distinct. Having the controller depend on an internal package in the scheduler means the two can never be separated. This is a bad outcome. We should not make this change.

hmmm.... Tests passed.  I'm wondering if this is a transitive dep and this file just lists all possible transitives you can rely upon.

Reply:
I think we can remove this now post recent rebases. Will force push an update if local builds go through

Reply:
@deads2k - You're right, `k8s.io/kubernetes/pkg/controller/daemon` has a dependency on `pkg/scheduler/algorithm/predicates which in turn has a dependency on `k8s.io/kubernetes/pkg/scheduler/internal`(or the util before renaming). So, we should be ok with this but we need to see if we can refactor the predicates into a more neutral place that so both scheduler and DS controller can reuse them so that we can remove the dependecy

Reply:
/cc @bsalamat

Reply:
> we should be ok with this but we need to see if we can refactor the predicates into a more neutral place that so both scheduler and DS controller can reuse them so that we can remove the dependency

we should do that *before* moving them into internal, rather than moving in a direction we don't want, then recovering

Reply:
FYI I've discussed `.import-restrictions` with @lavalamp on past PRs and it's true that they list transitive dependencies. My opinion is that this is a quirk of the tool, and transitive dependencies on `pkg/scheduler/internal` are no worse than direct dependencies on `pkg/scheduler`

There are various k8s components that import bits of the scheduler code base, somewhat ad hoc. Moving code into our internal folder is a way to draw a line in the sand and prevent more packages from leaking into our supported API surface.

I also hope that getting explicit about which libraries are reasonable to use in other components will lead to better APIs and separation of concerns down the road.